src/log.cpp

```diff
@@ -1,27 +1,27 @@
 #include "log.h"
 
-spdlog::level::level_enum get_log_level() {
-    const char* log_level_env = std::getenv("INF_LOGLEVEL");
-    if (log_level_env) {
-        std::string log_level_str(log_level_env);
-        if (log_level_str == "debug") {
-            return spdlog::level::debug;
-        } else if (log_level_str == "info") {
-            return spdlog::level::info;
-        } else if (log_level_str == "warn") {
-            return spdlog::level::warn;
-        } else if (log_level_str == "error") {
-            return spdlog::level::err;
-        } else if (log_level_str == "critical") {
-            return spdlog::level::critical;
-        }
-    }
-    return spdlog::level::info;
-}
-
 static auto console = spdlog::stdout_color_mt(APP_NAME);
 
-static bool log_level_set = []() {
-    console->set_level(get_log_level());
-    return true;
-}();
+void set_log_level(std::string level) {
+    if (level == "info") {
+        console->set_level(spdlog::level::info);
+    } else if (level == "error") {
+        console->set_level(spdlog::level::err);
+    } else if (level == "debug") {
+        console->set_level(spdlog::level::debug);
+    } else if (level == "warning") {
+        console->set_level(spdlog::level::warn);
+    } 
+}
+
+void printing(std::string level, std::string msg) {
```

Comment:
since we already have set_log_level, why we want to pass level in each printing? can we  just set_log_level once, and use printing(maybe have other name) without logleve?

Reply:
set_log_level is for the infinistore log level config. this level is from user perspective, which level user wants to print the message